src/rgw/rgw_data_sync.cc

```diff
@@ -5221,73 +5180,97 @@ int RGWSyncBucketCR::operate(const DoutPrefixProvider *dpp)
     do {
       tn->log(20, SSTR("sync status for source bucket: " << bucket_status.state));
 
-      // if the state wasn't Incremental, take a bucket-wide lease to prevent
-      // different shards from duplicating the init and full sync
-      if (bucket_status.state != BucketSyncState::Incremental) {
-        assert(!bucket_lease_cr);
-        bucket_lease_cr.reset(new RGWContinuousLeaseCR(env->async_rados, env->store, status_obj,
-                                                       lock_name, lock_duration, this));
-        yield spawn(bucket_lease_cr.get(), false);
-        while (!bucket_lease_cr->is_locked()) {
-          if (bucket_lease_cr->is_done()) {
-            tn->log(5, "failed to take bucket lease");
-            set_status("lease lock failed, early abort");
-            drain_all();
-            return set_cr_error(bucket_lease_cr->get_ret_status());
-          }
-          tn->log(5, "waiting on bucket lease");
-          yield set_sleeping(true);
-        }
-        // reread the status after acquiring the lock
-        yield call(new ReadCR(dpp, env->async_rados, env->svc->sysobj,
-                              status_obj, &bucket_status, false, &objv));
+
+      if (bucket_status.state == BucketSyncState::Init ||
+          bucket_status.state == BucketSyncState::Stopped) { 
+        // if state is Init or Stopped, we query the remote RGW for ther state
+		    yield call(new RGWReadRemoteBucketIndexLogInfoCR(sc, sync_pair.dest_bs.bucket, &info));
         if (retcode < 0) {
-          bucket_lease_cr->go_down();
-          drain_all();
-          bucket_lease_cr.reset();
           return set_cr_error(retcode);
         }
       }
 
-      if (bucket_status.state == BucketSyncState::Init ||
-          bucket_status.state == BucketSyncState::Stopped) {
-        assert(bucket_lease_cr);
+      if (bucket_status.state != BucketSyncState::Incremental)  {
```

Comment:
this block is calling `InitBucketFullSyncStatusCR` when `bucket_status.state == BucketSyncState::Full` where it didn't before